### PR TITLE
Refactor everything

### PR DESCRIPTION
**Pull request by Scrumplex**
Move to pydantic models.
Cleanup whole codebase.
Make it easier to run the scripts individually.

This version doesn't produce the exact same result as before, as some small details changed (timestamps now always include TZ info, some fields were omitted if they were unchanged, now they only will be omitted if they are None / null)

### CODE REVIEW

File: generateLiteloader.py [hunk 1]
```diff
@@ -1,93 +1,108 @@
...
+def process_artefacts(mc_version: str, artefacts: Dict[str, LiteloaderArtefact], is_snapshot: bool) \
+        -> Tuple[List[MetaVersion], Optional[MetaVersion]]:
+    versions: List[MetaVersion] = []
+    lookup: Dict[str, MetaVersion] = {}
+    latest_version = None
     latest = None
-    for id, artefact in liteloader.items():
-        if id == 'latest':
-            latestVersion = artefact.version
+    for x, artefact in artefacts.items():
```
Comment (txtsd) [inline generateLiteloader.py:23]: Refactor the `artefact` variables and functions to `artifact` pls?

Comment (Scrumplex) [reply generateLiteloader.py:23]: LiteLoader calls it "Artefact" which is why we also use the same names here.

Comment (Scrumplex) [reply generateLiteloader.py:23]: Though we could rename all of that to artifact and define `artefact` as an alias in the model

**Comment by txtsd** [comment]
I can't push to this, can I?

**Comment by Scrumplex** [comment]
> I can't push to this, can I?

You should be able to push. It's just a normal branch directly on this repo

File: generateLiteloader.py (hunk 1 above)
Comment (txtsd) [reply generateLiteloader.py:23]: Oh it's `artefact` in the jsons so no worries

**Review by txtsd — APPROVED** [review]
LGTM!

**Pull request merged**